pkg/k8s/apps/crud_test.go

```diff
@@ -543,3 +544,337 @@ func TestListDevModeOn(t *testing.T) {
+											v1.ResourceCPU:    mustParseK8sQuantity("100m"),
+										},
+										Limits: v1.ResourceList{
+											v1.ResourceMemory: mustParseK8sQuantity("256Mi"),
+											v1.ResourceCPU:    mustParseK8sQuantity("200m"),
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedResources: model.ResourceRequirements{
+				Requests: model.ResourceList{
+					v1.ResourceMemory: mustParseModelQuantity("128Mi"),
+					v1.ResourceCPU:    mustParseModelQuantity("100m"),
+				},
+				Limits: model.ResourceList{
+					v1.ResourceMemory: mustParseModelQuantity("256Mi"),
+					v1.ResourceCPU:    mustParseModelQuantity("200m"),
+				},
+			},
+			shouldInherit: true,
+		},
+		{
+			name:        "do not inherit when env var is false",
+			envVarValue: "false",
+			dev: &model.Dev{
+				Name:      "test",
+				Container: "test",
+				Resources: model.ResourceRequirements{},
+			},
+			deployment: &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test",
+					Namespace: "test",
+				},
+				Spec: appsv1.DeploymentSpec{
+					Template: v1.PodTemplateSpec{
+						Spec: v1.PodSpec{
+							Containers: []v1.Container{
+								{
+									Name: "test",
+									Resources: v1.ResourceRequirements{
+										Requests: v1.ResourceList{
+											v1.ResourceMemory: mustParseK8sQuantity("128Mi"),
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedResources: model.ResourceRequirements{},
+			shouldInherit:     false,
+		},
+		{
+			name:        "do not inherit when dev resources are not empty",
+			envVarValue: "true",
+			dev: &model.Dev{
+				Name:      "test",
+				Container: "test",
+				Resources: model.ResourceRequirements{
+					Requests: model.ResourceList{
+						v1.ResourceMemory: mustParseModelQuantity("64Mi"),
+					},
+				},
+			},
+			deployment: &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test",
+					Namespace: "test",
+				},
+				Spec: appsv1.DeploymentSpec{
+					Template: v1.PodTemplateSpec{
+						Spec: v1.PodSpec{
+							Containers: []v1.Container{
+								{
+									Name: "test",
+									Resources: v1.ResourceRequirements{
+										Requests: v1.ResourceList{
+											v1.ResourceMemory: mustParseK8sQuantity("128Mi"),
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedResources: model.ResourceRequirements{
+				Requests: model.ResourceList{
+					v1.ResourceMemory: mustParseModelQuantity("64Mi"),
+				},
+			},
+			shouldInherit: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Save original environment variable value
+			originalValue := os.Getenv(model.OktetoInheritKubernetesResourcesEnvVar)
+			defer func() {
+				if originalValue == "" {
+					os.Unsetenv(model.OktetoInheritKubernetesResourcesEnvVar)
+				} else {
+					os.Setenv(model.OktetoInheritKubernetesResourcesEnvVar, originalValue)
+				}
```

Comment:
Let's use `t.setenv`, so we don't have to explicitly clean it up. Also, let's try to remove the if/else